### PR TITLE
Adds OTEL tests, fixes some issues with propagation

### PR DESCRIPTION
**Pull request by ksh-chy**
## Why

There were some small issues with propagation due to how we associate context's with spans. When we create a span, we need to ensure that we use `trace.setSpan` to associate the span with a context. The `trace.startSpan` function accepts a context as well, but this is meant to be used to access the parent span, and the context is not updated by itself. 

## What changed

Add some dev dependencies for testing, updated functions in `tracing/index.ts` to follow the convention above

## Versioning

- [ ] Breaking protocol change
- [ ] Breaking ts/js API change

### CODE REVIEW

File: util/testHelpers.ts [hunk 1]
```diff
@@ -24,6 +35,18 @@ import { Connection, Session, SessionOptions } from '../transport/session';
...
+  const provider = new BasicTracerProvider();
+  provider.addSpanProcessor(
+    new SimpleSpanProcessor(new InMemorySpanExporter()),
+  );
+  const contextManager = new StackContextManager();
+  contextManager.enable();
+  trace.setGlobalTracerProvider(provider);
+  context.setGlobalContextManager(contextManager);
+  propagation.setGlobalPropagator(new W3CTraceContextPropagator());
+});
```
Comment (jackyzha0) [inline util/testHelpers.ts:48]: seems weird to do this at a global level, can we do this just for tracing tests?

**Review by jackyzha0 — APPROVED** [review]

File: util/testHelpers.ts (hunk 1 above)
Comment (ksh-chy) [reply util/testHelpers.ts]: Done!

File: tracing/tracing.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { W3CTraceContextPropagator } from '@opentelemetry/core';
+import { StackContextManager } from '@opentelemetry/sdk-trace-web';
+
+import tracer from './index';
+import * as tracing from './index';
+import { OpaqueTransportMessage } from '../transport';
+
+beforeAll(() => {
```
Comment (jackyzha0) [inline tracing/tracing.test.ts:19]: sorry last nit: does it still work if we move it inside the describe block?

Comment (ksh-chy) [reply tracing/tracing.test.ts:19]: Yeah that seems to be fine, let me commit this change

**Pull request merged**